### DISCUSSION TITLE
Laravel Vapor - SQS queue

### DISCUSSION THREAD

**Discussion by michalbursik** [spatie/laravel-event-sourcing#427] [Q&A]
Hello,

According to Laravel [Vapor docs](https://docs.vapor.build/resources/queues.html): **In addition, the concurrency value must be an integer between 2 and 1,000.**

According to this [spatie/laravel-event-sourcing docs](https://spatie.be/docs/laravel-event-sourcing/v7/installation-setup): **You should make sure that the queue will process only one job at a time.**

I also tried the database driver with default config (except queue = env('EVENT_PROJECTOR_QUEUE_NAME') on localhost, but with no luck. Basically:

```php
    public static function createWithAttributes(array $attributes): Account
    {
        $attributes['uuid'] = (string) Uuid::uuid4();

        event(new AccountCreated($attributes));

        return static::uuid($attributes['uuid']); # Returns null, because the event is not processed synchronously.
    }
```

Is there a way to setup event sourcing package with Laravel Vapor and SQS queue, so that jobs are executing synchronously?

Can you help me out ?

Thanks

**Comment by Sparclex** [comment]
I was wondering, did you manage to make it work?

**Reply by michalbursik** [reply]
Unfortunately, no.

**Comment by nastal** [comment]
I believe you can you use WithoutOverlapping middleware, what will uses aggregate uuid as key, if by default vapor offers max processes at queue. It works with atomic cache locks.
https://laravel.com/docs/11.x/queues#preventing-job-overlaps

As also, there was a discussion about race condition on persisting on event store. Now it supports unique composite key on aggregate uuid and version. And you can try to use retry middleware on command bus, if exception on unique key was catched in concurrent threads.
